biodatasets/pubtator_central/pubtator_central.py

```diff
@@ -0,0 +1,317 @@
+  author       = {Wei, Chih-Hsuan and Allot, Alexis and Leaman, Robert and Lu, Zhiyong},
+  year         = 2019,
+  month        = {05},
+  journal      = {Nucleic Acids Research},
+  volume       = 47,
+  number       = {W1},
+  pages        = {W587-W593},
+  doi          = {10.1093/nar/gkz389},
+  issn         = {0305-1048},
+  url          = {https://doi.org/10.1093/nar/gkz389},
+  eprint       = {https://academic.oup.com/nar/article-pdf/47/W1/W587/28880193/gkz389.pdf}
+}
+"""
+
+_DATASETNAME = "pubtator_central"
+
+_DESCRIPTION = """\
+PubTator Central (PTC, https://www.ncbi.nlm.nih.gov/research/pubtator/) is a web service for
+exploring and retrieving bioconcept annotations in full text biomedical articles. PTC provides
+automated annotations from state-of-the-art text mining systems for genes/proteins, genetic
+variants, diseases, chemicals, species and cell lines, all available for immediate download. PTC
+annotates PubMed (30 million abstracts), the PMC Open Access Subset and the Author Manuscript
+Collection (3 million full text articles). Updated entity identification methods and a
+disambiguation module based on cutting-edge deep learning techniques provide increased accuracy.
+"""
+
+_HOMEPAGE = "https://www.ncbi.nlm.nih.gov/research/pubtator/"
+
+# TODO: Add the licence for the dataset here (if possible)
+# Note that this doesn't have to be a common open source license.
+# Some datasets have custom licenses. In this case, simply put the full license terms
+# into `_LICENSE`
+_LICENSE = ""
+
+_URLS = {
+    "api": "https://www.ncbi.nlm.nih.gov/research/pubtator-api/publications/export/pubtator",
+    "source": "https://ftp.ncbi.nlm.nih.gov/pub/lu/PubTatorCentral/bioconcepts2pubtatorcentral.offset.gz",
+    "bigbio_kb": "https://ftp.ncbi.nlm.nih.gov/pub/lu/PubTatorCentral/bioconcepts2pubtatorcentral.offset.gz",
+}
+
+_SUPPORTED_TASKS = [Tasks.NAMED_ENTITY_RECOGNITION, Tasks.NAMED_ENTITY_DISAMBIGUATION]
+
+_SOURCE_VERSION = "2022.01.08"
+
+_BIGBIO_VERSION = "1.0.0"
+
+
+class PubtatorCentralDataset(datasets.GeneratorBasedBuilder):
+    """PubTator Central"""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        BigBioConfig(
+            name="pubtator_central_api",
+            version=SOURCE_VERSION,
+            description="PubTator Central API schema",
+            schema="api",
+            subset_id="pubtator_central",
+        ),
+        BigBioConfig(
+            name="pubtator_central_source",
+            version=SOURCE_VERSION,
+            description="PubTator Central source schema",
+            schema="source",
+            subset_id="pubtator_central",
+        ),
+        BigBioConfig(
+            name="pubtator_central_bigbio_kb",
+            version=BIGBIO_VERSION,
+            description="PubTator Central BigBio schema",
+            schema="bigbio_kb",
+            subset_id="pubtator_central",
+        ),
+    ]
```

Comment:
I didn't mean to discourage use of the full dataset here. I think it would be great to have both the `sample` dataset and the `full` dataset as options (so still four configs but now the API is replaced with the sample)